### PR TITLE
Upgrade to 1.6.2-1

### PR DESCRIPTION
**Pull request by utay**
Context in https://github.com/frontapp/front/pull/24480 and https://github.com/frontapp/front/pull/24496#discussion_r586373401.

TL;DR:
3-years-old issues addressed by our fork have been fixed (with other small bugs and way better error handling) on master so we could use `express-http-proxy` directly but there's a new issue (https://github.com/villadora/express-http-proxy/issues/359) which makes GET requests to CloudFront/S3 fail because they have `{}` as the body. We can use this PR https://github.com/villadora/express-http-proxy/pull/457 on the original repo to fix it (by not parsing the body if `req.method === 'GET'`) but it hasn't been merged yet and looking at the maintainer activity, I think it won't be merged in the next weeks. So I ended up taking the latest master and applying this PR as a patch in our fork. 

Steps done to upgrade + get patch from https://github.com/villadora/express-http-proxy/pull/457:
- `git merge -X theirs villadora master`
- `git merge v-electrolux master`
- Change name and version in `package.json`

### CODE REVIEW

**Comment by seanxiesx** [comment]
lgtm. would do a diff with upstream as a quick sanity check after you merge

**Comment by utay** [comment]
> lgtm. would do a diff with upstream as a quick sanity check after you merge

Good point. To make sure, I just ran in this repo:
```
for file in $(find . -type f -name '*.js' ); do
  diff $file path-to-upstream/express-http-proxy/$file
done
```

<details>
<summary>
Diff output
</summary>

```
39d38
<     var parseReqBody = (typeof options.parseReqBody === 'function') ? options.parseReqBody(req) : options.parseReqBody;
41c40
<     if (parseReqBody) {
---
>     if (options.parseReqBody) {
12,13c12
<   var parseReqBody = (typeof options.parseReqBody === 'function') ? options.parseReqBody(req) : options.parseReqBody;
<   var parseBody = (!parseReqBody) ? Promise.resolve(null) : requestOptions.bodyContent(req, res, options);
---
>   var parseBody = (!options.parseReqBody) ? Promise.resolve(null) : requestOptions.bodyContent(req, res, options);
67c67
<   describe('when user sets parseReqBody as bool', function () {
---
>   describe('when user sets parseReqBody', function () {
143,186d142
<   describe('when user sets parseReqBody as function', function () {
<     it('should not parse body with form-data content', function (done) {
<       var filename = os.tmpdir() + '/express-http-proxy-test-' + (new Date()).getTime() + '-png-transparent.png';
<       var app = express();
<       app.use(proxy('localhost:8109', {
<         parseReqBody: (proxyReq) => proxyReq.headers['content-type'].includes('application/json'),
<         proxyReqBodyDecorator: function (bodyContent) {
<           assert(!bodyContent, 'body content should not be parsed.');
<           return bodyContent;
<         }
<       }));
<
<       fs.writeFile(filename, pngData, function (err) {
<         if (err) { throw err; }
<         request(app)
<           .post('/post')
<           .attach('image', filename)
<           .end(function (err) {
<             fs.unlinkSync(filename);
<
<             done(err);
<           });
<       });
<     });
<     it('should parse body with json content', function (done) {
<       var app = express();
<       app.use(proxy('localhost:8109', {
<         parseReqBody: (proxyReq) => proxyReq.headers['content-type'].includes('application/json'),
<         proxyReqBodyDecorator: function (bodyContent) {
<           assert(bodyContent, 'body content should be parsed.');
<           return bodyContent;
<         }
<       }));
<
<       request(app)
<         .post('/post')
<         .send({ some: 'json' })
<         .end(function (err) {
<           done(err);
<         });
<     });
<   });
<
<
89d88
<   parseReqBody = (typeof parseReqBody === 'function') ? parseReqBody(req) : options.parseReqBody;
```
</details>

Looks good!
